src/firefly/js/drawingLayers/ComputeWebGridData.js

```diff
@@ -748,31 +695,40 @@ function drawLabeledPolyLine (drawData, bounds, nLevel0,  label, labelLocations,
             ipt0= makeImageWorkSpacePt(x[i],y[i]);
             ipt1= makeImageWorkSpacePt(x[i+1], y[i+1]);
             if (!aitoff  ||  ((Math.abs(ipt1.x-ipt0.x) <screenWidth /8 ) && (aitoff))) {
-                drawData.push(ShapeDataObj.makeLine(ipt0,ipt1));
+                drawData.push(ShapeDataObj.makeLine(ipt0, ipt1));
+
+                //find the middle point of the line, index from 0, so minus 1
+                if (i===Math.round(x.length/2)-1 ) {
+
+                    var wpt1 = cc.getScreenCoords(ipt0);
+                    var wpt2 = cc.getScreenCoords(ipt1);
+                    const slope = (wpt2.y - wpt1.y) / (wpt2.x - wpt1.x);
+                    slopAngle = Math.atan(slope) * 180 / Math.PI;
```

Comment:
You could use `cc.getDeviceCoords` to get the real slope  after image rotation. I think that would fix you rotation problem.